example/glue/ObjectModel.hpp

```diff
@@ -80,32 +105,65 @@ class GC_ObjectModel
 	}
 
 	/**
+	 * Returns TRUE if an object is indexable, FALSE otherwise.
+	 *
+	 * @param objectPtr pointer to the object
+	 * @return TRUE if object is indexable, FALSE otherwise
+	 */
+	MMINLINE bool
+	isIndexable(omrobjectptr_t objectPtr)
+	{
+		return false;
+	}
+
+	/**
+	 * Set flags where ever they are
+	 * @param objectPtr Pointer to an object
+	 * @param bitsToClear mask to clear bits
+	 * @param bitsToSet mask to set bits
+	 */
+	MMINLINE void
+	setFlags(omrobjectptr_t objectPtr, uintptr_t bitsToClear, uintptr_t bitsToSet)
+	{
+		fomrobject_t* flagsPtr = (fomrobject_t*)SCAVENGER_FORWARDING_SLOT_EA(objectPtr);
```

Comment:
GC_ObjectModel::isIndexable() and GC_ObjectModel::setFlags() are generic
functions that depend on OMR_OBJECT_HEADER_SLOT_OFFSET (formerly
SCAVENGER_FORWARDING_SLOT_OFFSET). They are not Scavenger-specific.



From:	Charlie Gracie <notifications@github.com>
To:	eclipse/omr <omr@noreply.github.com>
Cc:	Kim T Briggs/Ottawa/IBM@IBMCA
Date:	04/18/2016 08:12 AM
Subject:	Re: [eclipse/omr] FIX #51 : Merge generational GC (gencon) into
            OMR (#65)



In example/glue/ObjectModel.hpp:


> +		 MMINLINE bool
> +		 isIndexable(omrobjectptr_t objectPtr)
> +		 {
> +		 		 return false;
> +		 }
> +
> +		 /**
> +		  * Set flags where ever they are
> +		  * @param objectPtr Pointer to an object
> +		  * @param bitsToClear mask to clear bits
> +		  * @param bitsToSet mask to set bits
> +		  */
> +		 MMINLINE void
> +		 setFlags(omrobjectptr_t objectPtr, uintptr_t bitsToClear,
uintptr_t bitsToSet)
> +		 {
> +		 		 fomrobject_t* flagsPtr =
(fomrobject_t*)SCAVENGER_FORWARDING_SLOT_EA(objectPtr);



Is this scavenger specific code?


—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub